spec.html

```diff
@@ -24178,16 +24392,359 @@ <h1>Runtime Semantics: Evaluation</h1>
     </emu-clause>
   </emu-clause>
 
+  <emu-clause id="sec-decorators">
+    <h1>Decorators</h1>
+    <h2>Syntax</h2>
+    <emu-grammar type="definition">
+      DecoratorList[Yield, Await] :
+        DecoratorList[?Yield, ?Await]? Decorator[?Yield, ?Await]
+
+      Decorator[Yield, Await] :
+        `@` DecoratorMemberExpression[?Yield, ?Await]
+        `@` DecoratorParenthesizedExpression[?Yield, ?Await]
+        `@` DecoratorCallExpression[?Yield, ?Await]
+
+      DecoratorMemberExpression[Yield, Await] :
+        IdentifierReference[?Yield, ?Await]
+        DecoratorMemberExpression[?Yield, ?Await] `.` IdentifierName
+
+      DecoratorParenthesizedExpression[Yield, Await] :
+        `(` Expression[+In, ?Yield, ?Await] `)`
+
+      DecoratorCallExpression[Yield, Await] :
+        DecoratorMemberExpression[?Yield, ?Await] Arguments[?Yield, ?Await]
+        DecoratorParenthesizedExpression[?Yield, ?Await] Arguments[?Yield, ?Await]
+    </emu-grammar>
+
+    <emu-clause id="sec-decoratorevaluation" type="sdo">
+      <h1>Runtime Semantics: DecoratorEvaluation</h1>
+      <emu-grammar>Decorator : `@` DecoratorMemberExpression</emu-grammar>
+      <emu-alg>
+        1. Let _expr_ be the |MemberExpression| that is covered by |DecoratorMemberExpression|.
+        1. Let _ref_ be the result of evaluating _expr_.
+        1. Return ? GetValue(_ref_).
+      </emu-alg>
+      <emu-grammar>Decorator : `@` DecoratorCallExpression</emu-grammar>
+      <emu-alg>
+        1. Let _expr_ be the |CallMemberExpression| that is covered by |DecoratorCallExpression|.
+        1. Let _ref_ be the result of evaluating _expr_.
+        1. Return ? GetValue(_ref_).
+      </emu-alg>
+      <emu-grammar>Decorator : `@` DecoratorParenthesizedExpression</emu-grammar>
+      <emu-alg>
+        1. Let _expr_ be the |MemberExpression| that is covered by |DecoratorParenthesizedExpression|.
+        1. Let _ref_ be the result of evaluating _expr_.
+        1. Return ? GetValue(_ref_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-decoratorlistevaluation" type="sdo">
+      <h1>Runtime Semantics: DecoratorListEvaluation</h1>
+      <emu-grammar>DecoratorList : DecoratorList? Decorator</emu-grammar>
+      <emu-alg>
+        1. If |DecoratorList| is present, then
+          1. Let _leftValue_ be ? DecoratorListEvaluation of |DecoratorList|.
+        1. Else,
+          1. Let _leftValue_ be a new empty List.
+        1. Let _rightValue_ be ? DecoratorEvaluation of |Decorator|.
+        1. Prepend _rightValue_ to _leftValue_.
+        1. Return _leftValue_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createdecoratoraccessobject" type="abstract operation">
+      <h1>
+        CreateDecoratorAccessObject (
+          _kind_: *"field"*, *"accessor"*, *"method"*, *"getter"*, or *"setter"*,
```

Comment:
Why are the strings being used here instead of the spec enums?

Reply:
The kinds pass through a couple of different AOs, so each AO would have to implement getting the string from the kind which seemed a bit repetitive. I suppose we could also add an AO for getting the string from the kind though, I'll do that.

Reply:
Actually I see now that we only really use the strings in one place, so will just inline the conversion there.